### PR TITLE
Data Prepper documentation for the SNS to SQS fan out pattern in S3

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

Some users have hit an issue using Data Prepper with an SNS/SQS fan out. This can be overcome and this documentation change conveys what a user needs to do for it.

### Issues Resolved

N/A

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Naarcha-AWS — APPROVED** [review]

File: _data-prepper/common-use-cases/s3-logs.md [hunk 1]
```diff
@@ -171,6 +171,10 @@ We recommend that you have one SQS queue per Data Prepper pipeline. In addition,
 
 If you have multiple pipelines, you must create multiple SQS queues for each pipeline, even if both pipelines use the same S3 bucket.
 
+## Amazon SNS fanout pattern
 
+To meet the scale of logs produced in S3, some users require multiple SQS queues for their logs. You can use the [Amazon Simple Notification Service](https://docs.aws.amazon.com/sns/latest/dg/welcome.html) (SNS) to route event notifications from S3 to an SQS [fanout pattern](https://docs.aws.amazon.com/sns/latest/dg/sns-common-scenarios.html). Using SNS, all S3 event notifications go directly to a single SNS topic, where you can subscribe to multiple SQS queues.
+
+To make sure that Data Prepper can parse the event from the SNS topic directly, configure [raw message delivery](https://docs.aws.amazon.com/sns/latest/dg/sns-large-payload-raw-message-delivery.html) on the SNS to SQS subscription. If you want Data Prepper to ignore an SQS queue that sends notifications to your SNS topic, do not enable raw message delivery those queues.
```
Comment (dlvenable) [inline _data-prepper/common-use-cases/s3-logs.md:178]: > If you want Data Prepper to ignore an SQS queue

I don't think this properly conveys the situation. Data Prepper isn't ignoring these SQS queues. They are not being used at all by Data Prepper. Also, the user can configure those other queues to use raw message delivery or not. It doesn't affect Data Prepper and Data Prepper doesn't affect them.

The point of my original wording was to help with the objection that setting raw message delivery will affect other consumers.

Comment (Naarcha-AWS) [reply _data-prepper/common-use-cases/s3-logs.md:178]: How about this?

"When raw message delivery is not enabled on an SQS queue, Data Prepper does not use that SQS queue."

Comment (dlvenable) [reply _data-prepper/common-use-cases/s3-logs.md:178]: I'm still not sure if that really conveys what I'm trying to say here.

This is my original text:

> The raw message delivery option is on the connection between SNS and SQS.
Thus, you can have some SQS queues which do not enable raw message delivery.
Using Data Prepper will not interfere with those queues.

Perhaps we can simply say:

> Setting this option will not affect other SQS queues which are subscribed to that SNS topic.

File: _data-prepper/common-use-cases/s3-logs.md [hunk 2]
```diff
@@ -171,6 +171,10 @@ We recommend that you have one SQS queue per Data Prepper pipeline. In addition,
 
 If you have multiple pipelines, you must create multiple SQS queues for each pipeline, even if both pipelines use the same S3 bucket.
 
+## Amazon SNS fanout pattern
 
+To meet the scale of logs produced in S3, some users require multiple SQS queues for their logs. You can use the [Amazon Simple Notification Service](https://docs.aws.amazon.com/sns/latest/dg/welcome.html) (SNS) to route event notifications from S3 to an SQS [fanout pattern](https://docs.aws.amazon.com/sns/latest/dg/sns-common-scenarios.html). Using SNS, all S3 event notifications go directly to a single SNS topic, where you can subscribe to multiple SQS queues.
```
Comment (natebower) [inline _data-prepper/common-use-cases/s3-logs.md:176]:
```suggestion
To meet the scale of logs produced by S3, some users require multiple SQS queues for their logs. You can use [Amazon Simple Notification Service](https://docs.aws.amazon.com/sns/latest/dg/welcome.html) (Amazon SNS) to route event notifications from S3 to an SQS [fanout pattern](https://docs.aws.amazon.com/sns/latest/dg/sns-common-scenarios.html). Using SNS, all S3 event notifications are sent directly to a single SNS topic, where you can subscribe to multiple SQS queues.
```

File: _data-prepper/common-use-cases/s3-logs.md [hunk 3]
```diff
@@ -171,6 +171,10 @@ We recommend that you have one SQS queue per Data Prepper pipeline. In addition,
...
+
+To make sure that Data Prepper can parse the event from the SNS topic directly, configure [raw message delivery](https://docs.aws.amazon.com/sns/latest/dg/sns-large-payload-raw-message-delivery.html) on the SNS to SQS subscription. Setting this option will not affect other SQS queues which are subscribed to that SNS topic.
```
Comment (natebower) [inline _data-prepper/common-use-cases/s3-logs.md:178]:
```suggestion
To make sure that Data Prepper can directly parse the event from the SNS topic, configure [raw message delivery](https://docs.aws.amazon.com/sns/latest/dg/sns-large-payload-raw-message-delivery.html) on the SNS to SQS subscription. Setting this option will not affect other SQS queues that are subscribed to that SNS topic.
```

**Review by natebower — APPROVED** [review]
@Naarcha-AWS Approved with noted changes. Thanks!

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**